styles/Canonical/009-Headings-no-links.yml

```diff
@@ -0,0 +1,19 @@
+# 09 - Heading - Headings should not contain links
+
+extends: existence
+message: "Headings should not contain links"
+link: https://docs.ubuntu.com/styleguide/en/#other-considerations
+nonword: true
+level: warning
+scope: raw
+tokens:
+  - '[\#]+ [ \w]+\[[\w]+[ \w]*\]\([.\w]+\)'
+  - '[ \w]+\[[\w]+[ \w]*\]\([.\w]+\)[^\n]*\n[=-]{2,}\n'
+  - '[\#]+ [ \w]+<http[^\n]+>'
```

Comment:
With the `linkify` MyST extension turned on (which is the default in the starter pack), you don't need the `<`...`>`, and it will actually be converted to a link even in the heading.
Not sure if we want to check for this though. I haven't searched now, maybe it's possible to disable linkify for headings, that might be an easier solution than trying to catch all possible links otherwise.